### PR TITLE
Use ICU for message i18n templatization and pluralizaiton

### PR DESCRIPTION
**Pull request by warrengm**
First step in adopting Lighthouse's i18n framework. #87 

This change also ensures that plurals are translated correctly. Previously they were formatted in code, which would break in other languages.

### CODE REVIEW

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 1]
```diff
@@ -42,20 +42,20 @@
       "title": "Latency of first ad request",
       "failureTitle": "Reduce latency of first ad request",
       "description": "This metric measures the elapsed time from the start of page load until the first ad request is made. Delayed ad requests will decrease impressions and viewability, and have a negative impact on ad revenue. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/metrics).",
-      "displayValue": "%s s"
+      "displayValue": "{adReqTime, number, seconds} s"
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space to be consistent with line 20 above.

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 2]
```diff
@@ -42,20 +42,20 @@
...
     },
     "ad-request-from-tag-load": {
       "title": "Latency of first ad request, from tag load",
       "failureTitle": "Reduce latency of first ad request (from tag load)",
       "description": "This metric measures the elapsed time from when the Google Publisher Tag loads until the first ad request is made. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/metrics).",
-      "displayValue": "%s s"
+      "displayValue": "{adReqTime, number, seconds} s"
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 3]
```diff
@@ -42,20 +42,20 @@
...
     },
     "ad-top-of-viewport": {
       "title": "No ad found at the very top of the viewport",
       "failureTitle": "Move the top ad further down the page",
       "description": "Over 10% of ads are never viewed because users scroll past them before they become viewable. By moving ad slots away from the very top of the viewport, users are more likely to see ads before scrolling away. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/ad-top-of-viewport).",
       "displayValue": "",
-      "failureDisplayValue": "A scroll of %s px would hide half of your topmost ad",
+      "failureDisplayValue": "A scroll of {valueInPx, number} px would hide half of your topmost ad",
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: I kind of think the space before "px" should also be removed, to be consistent with CSS styling rules. But I don't feel super strongly about that.

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 4]
```diff
@@ -91,20 +91,20 @@
       "title": "Latency of first ad render",
       "failureTitle": "Reduce time to render first ad",
       "description": "This metric measures the time for the first ad iframe to paint from page navigation. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/metrics).",
-      "displayValue": "%s s"
+      "displayValue": "{firstAdPaint, number, seconds} s"
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 5]
```diff
@@ -91,20 +91,20 @@
...
       "displayValue": "",
-      "failureDisplayValue": "%s%% of viewport width is underutilized"
+      "failureDisplayValue": "{percentUnused, number, percent} of viewport width is underutilized"
     },
     "idle-network-times": {
       "title": "[Experimental] Network is efficiently utilized before ad requests",
       "failureTitle": "[Experimental] Reduce network idle time before ad requests",
       "description": "Moments of network idleness in the critical path to ad loading present opportunities to improve speed. Consider eliminating long tasks, timeouts, waiting on load events, or synchronous resources to avoid idleness. Chrome DevTools can be used to discover what is causing network idleness. [Learn more](https://developers.google.com/web/tools/chrome-devtools/evaluate-performance/reference).",
-      "displayValue": "%s s spent idle in critical path",
+      "displayValue": "{idleTime, number, seconds} s spent idle in critical path",
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 6]
```diff
@@ -160,7 +161,7 @@
       "title": "GPT tag is loaded statically",
       "failureTitle": "Load GPT tag statically",
       "description": "Tags loaded dynamically are not visible to the browser preloader. Consider using a static tag or `<link rel=\"preload\">` so the browser may load GPT sooner. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/static-ad-tags).",
-      "failureDisplayValue": "Up to %s s tag load time improvement",
+      "failureDisplayValue": "Up to {opportunity, number, seconds} s tag load time improvement",
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space

File: lighthouse-plugin-publisher-ads/messages/en-US.json [hunk 7]
```diff
@@ -170,14 +171,13 @@
       "title": "Tag load time",
       "failureTitle": "Reduce tag load time",
       "description": "This metric measures the time for the Google Publisher Tag's implementation script (pubads_impl.js) to load after the page loads. [Learn more](https://developers.google.com/publisher-ads-audits/reference/audits/metrics).",
-      "displayValue": "%s s"
+      "displayValue": "{tagLoadTime, number, seconds} s"
```
Comment (jonkeller) [inline lighthouse-plugin-publisher-ads/messages/en-US.json]: Nit: remove space

File: lighthouse-plugin-publisher-ads/messages/en-US.json (hunk 1 above)
Comment (warrengm) [reply lighthouse-plugin-publisher-ads/messages/en-US.json]: I'll add the space to line 20 for consistency with lighthouse core

**Pull request merged**